optuna/study/study.py

```diff
@@ -482,9 +537,10 @@ def tell(
         self,
         trial: Union[trial_module.Trial, int],
         values: Optional[Union[float, Sequence[float]]] = None,
-        state: TrialState = TrialState.COMPLETE,
+        state: Optional[TrialState] = None,
         skip_if_finished: bool = False,
-    ) -> None:
+        suppress_warning: bool = False,
```

Comment:
This is quite convenient. Also noticed you've added a test 👍  I'm wondering if we really want to expose this to the user however. It communicates complexity to an already somewhat complex method. Did you consider controlling this behavior by some internal method, e.g. `_tell` (although it's currently "used").

Reply:
It makes sense. I "restored" `_tell` (removing `@deprecated`) and moved the logic from `tell` to `_tell`.
https://github.com/optuna/optuna/compare/2d7ae11ef...b1b870f

Reply:
Thanks, I'll have a look. 🙇